### PR TITLE
tidb: add proxy-protocol-fallbackable

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

`proxy-protocol-fallbackable` is introduced in TiDB v6.5.1

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/41393

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: command-line-flags-for-tidb-configuration.md [hunk 1]
```diff
@@ -130,6 +130,11 @@ aliases: ['/docs-cn/dev/command-line-flags-for-tidb-configuration/','/docs-cn/de
 >
 > 如果使用 AWS 的 Network Load Balancer (NLB) 并开启 PROXY 协议，需要设置 NLB 的 `target group` 属性：将 `proxy_protocol_v2.client_to_server.header_place` 设为 `on_first_ack`。同时向 AWS 的 Support 提工单开通此功能的支持。注意，AWS NLB 在开启 PROXY 协议后，客户端将无法获取服务器端的握手报文，因此报文会一直阻塞到客户端超时。这是因为，NLB 默认只在客户端发送数据之后才会发送 PROXY 的报文，而在客户端发送数据包之前，服务器端发送的任何数据包都会在内网被丢弃。
 
+## `--proxy-protocol-fallbackable`
+
++ 用于控制是否启用 PROXY 协议回退模式。如果设置为 `true`，连接会在客户端没有发送 PROXY 协议头时，返回客户端的 IP 地址。
```
Comment (xhebox) [inline command-line-flags-for-tidb-configuration.md:135]: true, 则tidb可以接受非proxy-protocol. 或者说接受不发送proxy header的链接.

Comment (djshow832) [reply command-line-flags-for-tidb-configuration.md:135]: 如果设置为 `true`，则在 TiDB 开启 PROXY 协议后，即使客户端没有发送 PROXY 协议头，也能接受客户端连接；否则，在客户端没有发送 PROXY 协议头时，TiDB 不接受客户端连接。

Comment (Oreoxmt) [inline command-line-flags-for-tidb-configuration.md:135]:
```suggestion
+ 用于控制是否启用 PROXY 协议回退模式。如果设置为 `true`，则在 TiDB 开启 PROXY 协议后，即使客户端没有发送 PROXY 协议头，也能接受客户端连接。否则，在客户端没有发送 PROXY 协议头时，TiDB 不接受客户端连接。
```

File: command-line-flags-for-tidb-configuration.md [hunk 2]
```diff
@@ -130,6 +130,11 @@ aliases: ['/docs-cn/dev/command-line-flags-for-tidb-configuration/','/docs-cn/de
 >
 > 如果使用 AWS 的 Network Load Balancer (NLB) 并开启 PROXY 协议，需要设置 NLB 的 `target group` 属性：将 `proxy_protocol_v2.client_to_server.header_place` 设为 `on_first_ack`。同时向 AWS 的 Support 提工单开通此功能的支持。注意，AWS NLB 在开启 PROXY 协议后，客户端将无法获取服务器端的握手报文，因此报文会一直阻塞到客户端超时。这是因为，NLB 默认只在客户端发送数据之后才会发送 PROXY 的报文，而在客户端发送数据包之前，服务器端发送的任何数据包都会在内网被丢弃。
 
+## `--proxy-protocol-fallbackable`
+
++ 用于控制是否启用 PROXY 协议回退模式。如果设置为 `true`，则在 TiDB 开启 PROXY 协议后，即使客户端没有发送 PROXY 协议头，也能接受客户端连接。否则，在客户端没有发送 PROXY 协议头时，TiDB 不接受客户端连接。
```
Comment (Oreoxmt) [inline command-line-flags-for-tidb-configuration.md:135]:
```suggestion
+ 用于控制是否启用 PROXY 协议回退模式。如果设置为 `true`，TiDB 可以接受非 PROXY 协议规范或者没有发送 PROXY 协议头的客户端连接。默认情况下，TiDB 仅接受发送 PROXY 协议头的客户端连接。
```

**Comment by Oreoxmt** [comment]
@djshow832 @xhebox PTAL again

**Review by djshow832 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**